### PR TITLE
Connect Manage Sessions modal to API Client

### PR DESCRIPTION
**Pull request by e-wai**
## Notion ticket link
[Camp Overview: Delete Session Change, Change Session Capacity](https://www.notion.so/uwblueprintexecs/Camp-Overview-Delete-Session-Change-Session-Capacity-6d18cb517d61447ab691f8a7d37978f6)

## Implementation description

Outstanding pieces of UX work
- [ ] Save disabled if invalid input (debounce?)
- [ ] Tooltip explaining why state invalid

## Steps to test
1. Create sessions if needed
2. End to end flow testable -> Open "Manage Sessions" Modal, delete and/or edit capacities
3. Check that modal closes, state updates as expected. If you re-open the modal, your changes should be as expected

## What should reviewers focus on?
* Check comments on PR

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/middlewares/validators/campValidators.ts [hunk 1]
```diff
@@ -393,7 +393,12 @@ export const updateCampSessionsDtoValidator = async (
   res: Response,
   next: NextFunction,
 ) => {
-  const campSessions = req.body.updatedCampSessions;
+  const body = req.body.data;
```
Comment (e-wai) [inline backend/typescript/middlewares/validators/campValidators.ts:396]: cheating, but makes for cleaner frontend request

File: backend/typescript/services/implementations/__tests__/campService.test.ts [hunk 2]
```diff
@@ -3,14 +3,11 @@ import CampService from "../campService";
...
   CreateCampSessionsDTO,
-  CreateFormQuestionsDTO,
   UpdateCampSessionDTO,
   UpdateCampDTO,
-  CampDTO,
 } from "../../../types";
 import MgCampSession from "../../../models/campSession.model";
 import MgCamp from "../../../models/camp.model";
-import MgFormQuestion from "../../../models/formQuestion.model";
 import FileStorageService from "../fileStorageService";
```
Comment (e-wai) [inline backend/typescript/services/implementations/__tests__/campService.test.ts:11]: linter

File: frontend/src/components/pages/CampOverview/index.tsx [hunk 3]
```diff
@@ -70,6 +90,98 @@ const CampOverviewPage = (): JSX.Element => {
...
+      }
+    });
+
+    return requests;
+  };
+
+  const getManageSessionsResult = async (
+    deletedSessions: Set<string>,
+    updatedCapacities: Map<string, string>,
+  ): Promise<ManageSessionsPromisesResult> => {
```
Comment (e-wai) [inline frontend/src/components/pages/CampOverview/index.tsx:110]: This is an overcomplicated approach :/ it's a small improvement that the promises are not awaited in a chain, but rather using `Promise.all`

File: frontend/src/components/pages/CampOverview/index.tsx [hunk 4]
```diff
@@ -70,6 +90,98 @@ const CampOverviewPage = (): JSX.Element => {
...
+      (manageSessionsResult.deletedSessions &&
+        manageSessionsResult === false) ||
+      (manageSessionsResult.updatedSessions &&
+        manageSessionsResult.updatedSessions.length !== updatedCapacities.size);
+
+    if (!requestFailed) {
+      setTimeout(() => {
+        setSessionsDidChange(true);
+        setCurrentCampSession(0);
+      }, 500);
```
Comment (e-wai) [inline frontend/src/components/pages/CampOverview/index.tsx:165]: another reload trick, global state would be really nice

File: frontend/src/components/pages/CampOverview/index.tsx [hunk 5]
```diff
@@ -70,6 +89,95 @@ const CampOverviewPage = (): JSX.Element => {
 
   const openManageSessionsModal = () => onOpen();
```
Comment (dhmi19) [inline frontend/src/components/pages/CampOverview/index.tsx:90]: ik not part of this PR but can't we rename the original onOpen function to be used as openManageSessionsModal? 

see like 62, instead of `const { isOpen, onOpen, onClose } = useDisclosure();` we can do 
```
const { isOpen: manageSessionsIsOpen, onOpen: manageSessionsOnOpen, onClose: : manageSessionsOnClose } = useDisclosure();
```

File: frontend/src/components/pages/CampOverview/index.tsx [hunk 6]
```diff
@@ -70,6 +89,95 @@ const CampOverviewPage = (): JSX.Element => {
...
 
+  const getUpdateCampSessionsRequestArray = (
+    updatedCapacities: Map<string, string>,
+  ): Array<UpdateCampSessionsRequest> => {
+    const requests: Array<UpdateCampSessionsRequest> = [];
+
+    updatedCapacities.forEach((capacity, id) => {
+      const updatedCapacityExists = capacity.trim().length > 0;
```
Comment (dhmi19) [inline frontend/src/components/pages/CampOverview/index.tsx:98]: not too bothered but why not use a map function to build the array of Requests? honestly dealing with the Map<string, string> is a pain so if it's causing too much issues this works, otherwise, I can see a use of filter + map to build the array

**Review by dhmi19 — APPROVED** [review]
looks good. Some of the logic with the promises does seem a little convoluted considering the size of the requests but I guess it could be good for performance. I tested edit session capacity but didn't want to delete sessions lol. 

Left a few minor comments

File: frontend/src/components/pages/CampOverview/index.tsx (hunk 6 above)
Comment (e-wai) [reply frontend/src/components/pages/CampOverview/index.tsx:98]: Ehh not sure, I'll merge for now

**Comment by e-wai** [comment]
^ Rebasing + need to revisit transaction error handling
